install-dev/upgrade/php/ps_1770_update_charset.php

```diff
@@ -0,0 +1,81 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+function ps_1770_update_charset()
+{
+    $adminFilterTableExists = $adminFilterFilterIdExists = $moduleHistoryTableExists = $translationTableExists = false;
+
+    try {
+        $adminFilterTableExists = (bool) Db::getInstance()->executeS(
+            'SELECT count(*) FROM ' . _DB_PREFIX_ . 'admin_filter'
+        );
+        if ($adminFilterTableExists) {
+            $adminFilterFilterIdExists = (bool) Db::getInstance()->executeS(
+                'SELECT count(filter_id) FROM ' . _DB_PREFIX_ . 'admin_filter'
+            );
+        }
+        $moduleHistoryTableExists = (bool) Db::getInstance()->executeS(
```

Comment:
The checks on each table should be performed separately (different try/catch blocks) because let's say the first check on `admin_filter` fails because the table doesn't exist Then the following checks won't be performed and `module_history` and `translation` will be considered non existent